cfme/tests/infrastructure/test_scvmm_specific.py

```diff
@@ -30,6 +47,74 @@ def vm(provider, small_template):
     vm.cleanup_on_provider()
 
 
+@pytest.fixture
+def cfme_vhd(provider, appliance):
+    """ Given a stream from the appliance, gets the cfme vhd"""
+    stream = appliance.version.stream()
+    try:
+        url = '{}/'.format(conf.cfme_data["basic_info"]["cfme_images_url"][stream])
+    except KeyError:
+        pytest.skip("No such stream: {} found in cfme_data.yaml".format(stream))
+    # get image name
+    image_name = get_vhd_name(url)
+    if not image_name:
+        pytest.skip("No hyperv vhd image at {}".format(url))
+    # download the image to SCVMM library
+    provider.mgmt.download_file(url + image_name, image_name)
+    provider.mgmt.update_scvmm_library()
+    yield image_name
+
+
+def create_appliance(provider, vhd):
+    """ Create an appliance from the VHD provided on SCVMM """
+    # script to create template
+    version = ".".join(re.findall(r"\d+", vhd)[0:4])
+    template_name = "cfme-{}-template".format(version)
+    template_script = """
+        $networkName = "cfme2"
+        $templateName = "{template_name}"
+        $templateOwner = "{domain}\\{user}"
+        $maxMemorySizeMb = 12288
+        $startMemeorySizeMb = 4096
+        $minMemorySizeMb = 128
+        $cpuCount = 4
+        $srcName = "{image}"
+        $srcPath = "L:\\Library\\VHDs\\$srcName"
+        $dbDiskName = "Blank Disk - Large.vhd"
+        $dbDiskSrcPath = "L:\\Library\\VHDs\\$dbDiskName"
```

Comment:
I think this and some other values could be placed in yamls

Reply:
Good idea, I put some additional variables in the yamls